### PR TITLE
fix: cannot run workflow with non-configurable variable

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**

Bug: Workflow params not set overwritable will make valid failed.
Reason: The input variable map is overwirte to request when validate.

**Solution:**

Aviod overwrite the raw variables when validate run workflow input.

**Related Issue:**

#1560

### CODE REVIEW

File: pkg/dao/workflow.go [hunk 1]
```diff
@@ -394,16 +395,18 @@ func OverwriteWorkflowVariables(vars map[string]string, config types.WorkflowVar
...
 	}
 
 	for i := range config {
 		name, value := config[i].Name, config[i].Value
-		if _, ok := vars[name]; !ok {
-			vars[name] = value
+		if _, ok := mergedVars[name]; !ok {
+			mergedVars[name] = value
 		}
 	}
```
Comment (gitlawr) [inline pkg/dao/workflow.go:407]: Can we set these in L387-L392 so that we only loop config once?

Comment (alexcodelf) [reply pkg/dao/workflow.go]: updated

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by thxCode — APPROVED** [review]
LGTM

**Pull request merged**